modules/setting/path.go

```diff
@@ -171,6 +171,9 @@ func InitWorkPathAndCfgProvider(getEnvFn func(name string) string, args ArgWorkP
 
 	// only read the config but do not load/init anything more, because the AppWorkPath and CustomPath are not ready
 	InitCfgProvider(tmpCustomConf.Value)
+	if HasInstallLock(CfgProvider) {
+		ClearEnvConfigKeys() // if the instance has been installed, do not pass the environment variables to sub-processes
```

Comment:
Wasn't the docker user expect when they change the env and it will affect the Gitea's configuration?

Reply:
The `GITEA__` (double underline) environment variables are for config options only during startup/installation. I do not think any user should depend on these values, they do not affect the Gitea's installed process either.

And, the environment variables like `GITEA_xxx`  (only one underline) are not affected.

Reply:
So maybe we need to update the documentation?

Reply:
Which document? No document says it works like that.

Reply:
Or maybe we should mention these envs can only be used when installing?

Reply:
Until some users ask so and we know why they do so.

Reply:
I am not sure whether you mean this: 2a5d37e3714d3022f1d75e97c9dd2ef7e963a1ae

"and won't be passed into Gitea's sub-processes."